### ISSUE TITLE
”An unknown exception was thrown during indexing.“ while excuting python post processing

### ISSUE THREAD

**Issue by helloellite** [CoatiSoftware/Sourcetrail#1157]
## test code
![image](https://user-images.githubusercontent.com/17682610/109783213-b4ed4100-7c44-11eb-906d-3cbbe23d4f1a.png)

### key point
line 2 ends with '\r' while line 3 starts with '\r\n', and function f used an unreferenced symbol, which triggers ptyhon post processing.

## reason
```
std::vector<std::string> TextAccess::readFile(const FilePath& filePath)
{
        ........
	std::ifstream srcFile;
	srcFile.open(filePath.str());
       ........
       			safeGetline(srcFile, line);
       ......
}

std::istream& safeGetline(std::istream& is, std::string& t)
{
	......
	while (true)
	{
		int c = sb->sbumpc();
		switch (c)
		{
		case '\n':
			return is;
		case '\r':
			if (sb->sgetc() == '\n')
			{
				sb->sbumpc();
			}
			return is;
       ......
}
```
Given raw file content "\r\r\n", srcFile opens it as text file, which transfering the 2rd and 3rd character "\r\n" as "\n" . But safeGetline seems supposing it to be the raw content "\r\r\n". Dunplicated dealing of "\r\n" makes the empy line(3rd line in test code) missing, and line mismatch in python post processing.

## fix
open file as binary?
```
	std::ifstream srcFile;
 -	srcFile.open(filePath.str());
+      srcFile.open(filePath.str(), std::ios::binary | std::ios::in);
```

**Comment by mlangkabel** [comment]
Thanks for the detailed report. I managed to reproduce the issue on Windows. Looking into it now.

**Comment by mlangkabel** [comment]
This was quite tricky. The fix @helloellite suggested worked, but the real issue was that SourcetrailDB used some different code to record file content to the database. So we actually had to fix SourcetrailDB and update Sourcetrail to use a new version of the Python indexer.
Fixed with merge of pull request #1159.
